pkg/server/image_pull.go

```diff
@@ -495,11 +487,33 @@ func appendInfoHandlerWrapper(ref string) func(f containerdimages.Handler) conta
 						}
 						c.Annotations[targetRefLabel] = ref
 						c.Annotations[targetDigestLabel] = c.Digest.String()
-						c.Annotations[targetImageLayersLabel] = layers
+						c.Annotations[targetImageLayersLabel] = getLayers(ctx, targetImageLayersLabel, children[i:], labels.Validate)
```

Comment:
This seems like an extra change that doesn't really fix the problem, except maybe now we store less redundant data.

Reply:
"This" being changing from all layers to just children.

Reply:
Could you elaborate this? Doesn't calling `getLayers` on `children` avoids the problem of hitting the label limit (see L511) ?

Reply:
I was under the impression that the limit was per key, not total?

Reply:
Currently, the limit is per key. If we need to have the "total" label size limit, what should be the limit (instead of 4096 chars per key)?

Reply:
The way we have this now the root gets all the layers anyway, so the way we are doing `children[:i]` does not really fix the label size problem.

It is handled by truncating the values in `getLayers`, however.

So, 1.4.2 changed the behavior even for values that do not hit the limit.

Reply:
`getLayers` solves the label limit problem but `children[:i]` isn't for avoiding the label size limit. `containerd.io/snapshot/cri.image-layers` is currently a label for enabling snapshotter to pre-process(resolve) layers related to the target layer in parallel. So this is for optimization.

The reason why I do `children[:i]` here is otherwise `children[0:limit]` is always passed to the snapshotter and it can't get chance to pre-process `children[limit:]` layers.

However I agree with that we should find the better ways instead of `children[:i]`.